pkg/grpc/api.go

```diff
@@ -1219,26 +1221,32 @@ func (a *api) InvokeActor(ctx context.Context, in *runtimev1pb.InvokeActorReques
 	policyRunner := resiliency.NewRunnerWithOptions(ctx, policyDef,
 		resiliency.RunnerOpts[*invokev1.InvokeMethodResponse]{
 			Disposer: resiliency.DisposerCloser[*invokev1.InvokeMethodResponse],
+			Accumulator: func(r *invokev1.InvokeMethodResponse) {
+				if r == nil {
+					return
+				}
+
+				data, dataErr = r.RawDataFull()
+			},
 		},
 	)
 	resp, err := policyRunner(func(ctx context.Context) (*invokev1.InvokeMethodResponse, error) {
 		return a.UniversalAPI.Actors.Call(ctx, req)
 	})
+	if resp != nil {
+		defer resp.Close()
+	}
+
 	if err != nil && !errors.Is(err, actors.ErrDaprResponseHeader) {
 		err = status.Errorf(codes.Internal, messages.ErrActorInvoke, err)
 		apiServerLogger.Debug(err)
 		return response, err
 	}
 
-	if resp == nil {
```

Comment:
The problem is that here you are changing the behavior in case the response object is actually nil. This can happen when using the .NET SDK, but can happen in other cases too, and now the error is swallowed.

IMHO this fix should be applied in `pkg/actors` where we change how the actor is invoked. Sometime over the course of the last months the code that checked for the header included by the .NET SDK was changed, and now isn't checked anymore. Instead, it should return `ErrDaprResponseHeader` like you see being checked above